Dockerfile

```diff
@@ -0,0 +1,44 @@
+# -----Perform common operations-----
+FROM python:3.7-slim-bullseye as base
+
+ENV POETRY_VERSION=1.1.13
+
+# System deps:
+RUN pip install "poetry==$POETRY_VERSION"
+
+# Copy across your application code
+WORKDIR /appcode
+COPY poetry.lock pyproject.toml ./
+
+# Project initialization:
+RUN poetry config virtualenvs.create false \
+  && poetry install --no-interaction --no-ansi
+
+# Creating folders, and files for a project:
+COPY todo_app ./todo_app
+
+# -----Configure for local test-----
+FROM base as test
+ENV FLASK_ENV development
+ENV FLASK_DEBUG 1
+# Define an entrypoint, and default launch command
+ENTRYPOINT poetry run pytest; poetry run watchmedo shell-command \ 
```

Comment:
Iirc, you were on Windows so you saw that the file watching didn't really work. If you're interested in trying this again outside the devops course, [pytest-watch](https://pypi.org/project/pytest-watch/) has a `--poll` option for watching the files for changes. This works around the Windows/Docker inotify event issue.

Reply:
Working on a mac. I think the option worked, but I'll note that down. Thanks Hugh.